privacyidea/lib/smsprovider/SmppSMSProvider.py

```diff
@@ -0,0 +1,139 @@
+            sys_id = self.smsgateway.option_dict.get("SYSTEM_ID")
+            passwd = self.smsgateway.option_dict.get("PASSWORD")
+            s_addr_ton = int(self.smsgateway.option_dict.get("S_ADDR_TON"),16)
+            s_addr_npi = int(self.smsgateway.option_dict.get("S_ADDR_NPI"),16)
+            s_addr = self.smsgateway.option_dict.get("S_ADDR")
+            d_addr_ton = int(self.smsgateway.option_dict.get("D_ADDR_TON"),16)
+            d_addr_npi = int(self.smsgateway.option_dict.get("D_ADDR_NPI"),16)
+			
+        else:
+            smsc_host = self.config.get('SMSC_HOST')
+            smsc_port = self.config.get('SMSC_PORT')
+            sys_id = self.config.get('SYSTEM_ID')
+            passwd = self.config.get('PASSWORD')
+            s_addr_ton = int(self.config.get('S_ADDR_TON'),16)
+            s_addr_npi = int(self.config.get('S_ADDR_NPI'),16)
+            s_addr = self.config.get('S_ADDR')
+            d_addr_ton = int(self.config.get('D_ADDR_TON'),16)
+            d_addr_npi = int(self.config.get('D_ADDR_NPI'),16)
+
+        if smsc_host is None:
+            log.warning("can not submit message. SMSC_HOST is missing.")
+            raise SMSError(-1, "No SMSC_HOST specified in the provider config.")
+        else:
+            if smsc_port is None:
+                log.warning("can not submit message. SMSC_PORT is missing.")
+                raise SMSError(-1, "No SMSC_PORT specified in the provider config.")
+
+        #SMPP Part
+        client = None
+        error_key = None 
+        try:
+            client = smpplib.client.Client(smsc_host.encode("ascii"), smsc_port.encode("ascii"))
+            client.connect()
+            try:
+                client.bind_transmitter(system_id=sys_id.encode("ascii"), password=passwd.encode("ascii"))
+                client.send_message(source_addr_ton=s_addr_ton,source_addr_npi=s_addr_npi,source_addr=s_addr.encode("ascii"),dest_addr_ton=d_addr_ton,dest_addr_npi=d_addr_npi,destination_addr=phone.encode("ascii"),short_message=message.encode("ascii"))
+            except KeyError as inst:
+                error_key = inst.args[0]
+            finally:
+                pass
+        except:
+            error_key = "Bad connection string" 
+        finally:
+            if client:    
+                client.disconnect()
+		
+        if error_key is not None:
+            raise SMSError(error_key, "SMS could not be "
+                                          "sent: %s" % error_key)
+        return True
+
+
+    @classmethod
+    def parameters(cls):
+        """
+        Return a dictionary, that describes the parameters and options for the
+        SMS provider.
+        Parameters are required keys to values.
+
+        :return: dict
+        """
+		
+        params = {"options_allowed": False,
+                    "parameters": {
+                        "SMSC_HOST": {
+                            "required": True,
+                            "description": "SMSC Host IP"},
```

Comment:
The descriptions should be prepared for translation:

      _("SMSC Host IP")

(Also the descriptions below)

Reply:
OK